assets/src/containers/GradeDistribution.js

```diff
@@ -29,39 +35,102 @@ function GradeDistribution (props) {
   const { classes, disabled, courseId } = props
   if (disabled) return (<Error>Grade Distribution view is hidden for this course.</Error>)
 
-  const [loaded, error, gradeData] = useGradeData(courseId)
-  if (error) return (<Error>Something went wrong, please try again later.</Error>)
-  if (loaded && isObjectEmpty(gradeData)) return (<Error>No data provided.</Error>)
+  const [gradeLoaded, gradeError, gradeData] = useGradeData(courseId)
+  const [userSettingLoaded, userSetting] = useUserSetting(courseId, 'grade')
+  const [settingChanged, setSettingChanged] = useState(false)
+  const [showGrade, setShowGrade] = useState(false)
 
-  const buildGradeView = gradeData => {
+  useEffect(() => {
+    if (userSettingLoaded) {
+      if (isObjectEmpty(userSetting)) {
```

Comment:
```suggestion
      if (isObjectEmpty(userSetting.default)) {
```

Reply:
when the user doesn't have any set in DB the default is `{default: ""}`

Reply:
Okay, great. Is there ever a case where an empty object is provided?

Reply:
no